src/main/webapp/dev/js/common/datepicker.js

```diff
@@ -73,11 +73,24 @@ function prepareDatepickers() {
         showOtherMonths: true,
         gotoCurrent: true,
         defaultDate: yesterday,
-        maxDate: today,
+        maxDate: $('#startdate').val(),
         onSelect() {
             const newPublishDate = getMinDateForPublishDate($('#visibledate').datepicker('getDate'));
             $('#publishdate').datepicker('option', 'minDate', newPublishDate);
         },
+        onClose() {
+            const startDate = $('#startdate');
+            const currentVisibleDate = $('#visibledate');
+
+            if (startDate.val() === currentVisibleDate.val()) {
+                const startTime = $('#starttime');
+                const visibleTime = $('#visibletime');
+                // the visible time should be before the start time
+                if (parseInt(startTime.val(), 10) < parseInt(visibleTime.val(), 10)) {
```

Comment:
No need to include the radix for base 10 parsing.

Reply:
emm thats to satisfy esLint...

Reply:
Ohh oops it's okay then. Should add tests though.